File: c/1094.c
#include <stdio.h>
#include <string.h>
int main(){
    float n, i, e, total, s = 0, c=0, r=0;
    char a;
    scanf("%f", &n);
    for(i=0;i<n;i++){
        scanf("%f %c", &e, &a);
        if (a == 'C'){
            c+=e;
        }else if (a == 'S'){
            s+=e;
        }else{
            r+=e;
        }
    }
    total+=c+s+r;
    printf("Total: %.0f cobaias\n", total);
    printf("Total de coelhos: %.0f\n", c);
    printf("Total de ratos: %.0f\n", r);
    printf("Total de sapos: %.0f\n", s);
    printf("Percentual de coelhos: %.2f %%\n", (c*100)/total);
    printf("Percentual de ratos: %.2f %%\n", (r*100)/total);
    printf("Percentual de sapos: %.2f %%\n", (s*100)/total);
}

File: python/1094.py
e = int(input())
dict = {
    'C':0,
    'R':0,
    'S':0,
}
for i in range(0, e):
    n = input().split(' ')
    dict[n[1]] += int(n[0])
    
total = sum(dict.values())
print(f"Total: {total} cobaias")
print(f"Total de coelhos: {dict['C']}")
print(f"Total de ratos: {dict['R']}")
print(f"Total de sapos: {dict['S']}")
print("Percentual de coelhos: {:.2f} %".format((dict['C']*100)/total))
print("Percentual de ratos: {:.2f} %".format((dict['R']*100)/total))
print("Percentual de sapos: {:.2f} %".format((dict['S']*100)/total))